lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,243 @@
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
+            .optional_field(FILE, Kind::bytes(), None)
+            // Every field with an underscore (_) prefix will be treated as an additional field.
+            // Allowed characters in field names are any word character (letter, number, underscore), dashes and dots.
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
+        //
+    }
+}
+
+/// Deserializer that builds an `Event` from a byte frame containing a GELF log
+/// message.
+#[derive(Debug, Clone)]
+pub struct GelfDeserializer {
+    regex: Regex,
+}
+
+impl Default for GelfDeserializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GelfDeserializer {
+    /// Create a new GelfDeserializer
+    pub fn new() -> GelfDeserializer {
+        GelfDeserializer {
+            regex: Regex::new(r"^[\w\.\-]*$").unwrap(),
+        }
+    }
+
+    /// Returns a UTC DateTime from a numeric Value.
+    fn parse_timestamp(&self, val: &Value) -> DateTime<Utc> {
+        let mut secs = 0;
+        let mut nsecs = 0;
+        if val.is_f64() {
+            secs = f64::trunc(val.as_f64().unwrap()) as i64;
+            nsecs = f64::fract(val.as_f64().unwrap()) as u32;
+        } else if val.is_i64() {
+            secs = val.as_i64().unwrap();
+        } else if val.is_u64() {
+            secs = val.as_u64().unwrap() as i64;
+        }
+        let naive = NaiveDateTime::from_timestamp(secs, nsecs);
+        DateTime::<Utc>::from_utc(naive, Utc)
+    }
+
+    /// Attemps to parse an integer from a Value.
+    fn parse_number(&self, val: &Value) -> Result<i64, vector_core::Error> {
+        if val.is_number() {
+            let mut number = 0;
+            if val.is_f64() {
+                number = f64::round(val.as_f64().unwrap()) as i64;
+            } else if val.is_u64() {
+                number = val.as_u64().unwrap() as i64;
+            } else if val.is_i64() {
+                number = val.as_i64().unwrap();
+            }
+            Ok(number)
+        } else if val.is_string() {
+            match val.as_str().unwrap().parse::<i64>() {
+                Ok(number) => Ok(number),
+                Err(_) => match val.as_str().unwrap().parse::<f64>() {
+                    Ok(number) => Ok(f64::round(number) as i64),
+                    Err(_) => Err(format!(
+                        "Event field {} does not match GELF spec version {}: must be a number",
+                        VERSION, GELF_VERSION
+                    )
+                    .into()),
+                },
+            }
```

Comment:
Still working on making combinators come natural to me!

Reply:
The code you've written is perfectly readable. The only advantage of using the combinators is that you can keep the nesting level flat instead of needing to branch out ever more 😄